subiquity/server/controllers/oem.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2023 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import asyncio
+import logging
+from typing import Optional
+
+from subiquitycore.context import with_context
+
+from subiquity.common.apidef import API
+from subiquity.common.types import OEMResponse
+from subiquity.server.apt import OverlayCleanupError
+from subiquity.server.controller import SubiquityController
+from subiquity.server.types import InstallerChannels
+from subiquity.server.ubuntu_drivers import (
+    CommandNotFoundError,
+    get_ubuntu_drivers_interface,
+    )
+
+log = logging.getLogger('subiquity.server.controllers.oem')
+
+
+class OEMController(SubiquityController):
+
+    endpoint = API.oem
+
+    model_name = "oem"
+
+    def __init__(self, app) -> None:
+        super().__init__(app)
+        # At this point, the source variant has not been selected but it only
+        # has an impact if we're listing drivers, not OEM metapackages.
+        self.ubuntu_drivers = get_ubuntu_drivers_interface(self.app)
+
+        self.task: Optional[asyncio.Task] = None
```

Comment:
nitpick: a more verbose name on self.task might be nice, where we have one task we tend to accumulate more.

Reply:
Updated, thanks!